standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -4109,7 +4110,7 @@ private List<Partition> add_partitions_core(final RawStore ms, String catName,
             + TableName.getQualified(catName, dbName, tblName) +
             " does not exist");
       }
-
+      MTable mTable = getMS().ensureGetMTable(catName, dbName, tblName);
```

Comment:
nit: could we use something like `ObjectStore.convertToMTable` to get a `MTable` from `tbl` instance? For example, adding a  `MetaStoreServerUtils.convertToMTable(RawStore, Table)` instead of introducing the `ensureGetMTable` API in `RawStore`.